src/DefaultBuilder/src/StaticWebAssetsFileProvider.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore
+{
+    internal class StaticWebAssetsFileProvider : IFileProvider
+    {
+        private static readonly StringComparison FilePathComparison = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+            StringComparison.OrdinalIgnoreCase :
+            StringComparison.Ordinal;
+
+        public StaticWebAssetsFileProvider(string pathPrefix, string contentRoot)
+        {
+            BasePath = new PathString(pathPrefix.StartsWith("/") ? pathPrefix : "/" + pathPrefix);
+            InnerProvider = new PhysicalFileProvider(contentRoot);
+        }
+
+        public PhysicalFileProvider InnerProvider { get; }
+
+        public PathString BasePath { get; }
+
+        /// <inheritdoc />
+        public IDirectoryContents GetDirectoryContents(string subpath)
+        {
+            if (!StartsWithBasePath(subpath, out var physicalPath))
+            {
+                return NotFoundDirectoryContents.Singleton;
+            }
+            else
+            {
+                return InnerProvider.GetDirectoryContents(physicalPath);
```

Comment:
Avoid encoding again
```suggestion
                return InnerProvider.GetDirectoryContents(physicalPath.Value);
```